pythx/models/request/base.py

```diff
@@ -11,6 +11,8 @@ class BaseRequest(abc.ABC):
 
     @classmethod
     def validate(cls, candidate):
+        if cls.schema is None:
+            raise TypeError("Cannot use BaseRequest.validate without a schema")
```

Comment:
Good informative error but I think we could abstract away the extra info. This error will mostly like occur when a derived class tries to call `cls.validate(d)` without setting the schema.
I'm unsure if we should give reveal too much of our implementation details.
Rephrasing to reveal the most relevant info:
```
if cls.schema is None:
      raise TypeError("Cannot validate request without a schema. Please load schema")
```
What do you think?

Reply:
Good point. This error will occur when a domain model doesn't have a JSON body (e.g. logout responses or analysis status requests). Maybe we should also bring this in somehow? Otherwise the user will be confused if we ask them to load a schema even though there can't be one

Reply:
Would it be possible to override`validate` for those models then?

Reply:
Yes. That's always possible in Python, after all [we are all consenting adults here](https://mail.python.org/pipermail/tutor/2003-October/025932.html). :+1:

Reply:
Ohmy that's pretty different from what's taught in class :joy: (The encapsulation part I mean)